app/src/main/java/com/example/habitsmasher/LoginFragment.java

```diff
@@ -131,6 +131,8 @@ public void onClick(View view) {
      * @param password the user password
      */
     private void signInUserWithEmailAndPassword(String email, String password) {
+        PasswordEncrypt passwordEncrypt = new PasswordEncrypt();
+        password = passwordEncrypt.encrypt(password);
```

Comment:
Reason as to why I used Caesar Cipher: 

    I originally tried to find a way for firebase to do all the encrypting then that way nothing would have to be changed in our code but after searching the internet I was unable to find any results on how to do so.

    I then tried going with a more sophisticated encryption method that involved generating an encrypted string which often used some form of randomization. This worked well until a user tried to log in again. 

Due to how I understand this method works, it takes what the user enters(password) and compares it to what is in the database(encrypted password). The method that would have decrypted the password does not work here because firebase does not use that. So by using some form of randomization in encryption, this leaves the user unable to access their account.

Reply:
So if I understand this correctly, this is a **two-way encryption**? Meaning, if someone goes through and gets the encrypted version of the password from the database, can they **decrypt** the password easily? 

Ideally, we would want to use **one way** encryption, that way the password is hashed somehow, and the original is only able to be decrypted by its own user's private key or something of that sort.

Reply:
Yes, this would be two-way encryption, so theoretically yes, someone can decrypt the password if they got a hold of the key used. 

I would first like to point out, hashing and encryption/decryption are two different things. Hashing generates a value that cannot be reversed back to its original value. Encryption/decryption scrambles the value in a way where it is hard to decipher without the key used. 

I tried to follow this tutorial on encryption/decryption:
https://www.baeldung.com/java-aes-encryption-decryption

With some modification, I was able to get an encrypted value of the user's password but this method quoted above only takes in what the user enters into the editText, let's say **123456** and compares it to what's stored in the database. 

Now because we encrypted the password when the user registered, **123456** becomes some encrypted value which is great, however, when the user tries to log in again, this method essentially checks one by one in the database for user with a matching email and password. We will find the matching email eventually because the user is registered in this case, however, the passwords compared will be something like:
**123456**==hashed value of **123456**

These will not equal and the method will keep checking and find no matching account, and fail to login. 

There is a decrypt method, but we have no way to use it, because we cannot edit how the method goes one by one checks over the password. The decrypt method would only work if we could enter the password as one of its arguments.

Reply:
It's a shame that firebase doesn't offer this out of the box (although imagine the legal consequences if something went awry). This might be a question to ask Alex on Thursday to see if other groups figured out a library that does this out of the box. However, this does satisfy his requirement, so good job!

Reply:
Okay, so just making sure the login functionality works as intended still? The ask was just to encrypt the password in the db. If that is done, and the login still works, I'm good to merge this in!